src/main/java/core/lib/FieldMatch.java

```diff
@@ -0,0 +1,23 @@
+package core.lib;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Constraint(validatedBy = PasswordValidator.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldMatch {
+    String message() default "Passwords don't match!";
+    
+    String field();
+    
+    String fieldMatch();
+    
+    Class<?>[] groups() default {};
+    
+    Class<? extends Payload>[] payload() default {};
```

Comment:
```suggestion
    Class<? extends Payload>[] payload() default {};
    
    @Target({ElementType.TYPE})
    @Retention(RetentionPolicy.RUNTIME)
    @interface List {
        FieldMatch[] value();
    }
```
Did this work without this code?

Reply:
It works. This part is needed if there are several fields that need to be validated (like 2 passwords, 2 answers for secret questions and so on)